xsd-core/src/main/java/io/ballerina/xsd/core/Utils.java

```diff
@@ -143,4 +145,23 @@ static String extractTypeName(String[] values) {
         }
         return null;
     }
+
+    public static boolean isBallerinaKeyword(String fieldName) {
+        List<String> keywords = new ArrayList<>(Arrays.asList(
```

Comment:
I don't think it is a good idea to maintain a keywords list here. Shouldn't we get the list of keywords from a lang API? This will be hard to maintain otherwise.

Reply:
@heshanpadmasiri Is there any API for this we can use? I thought they are not publicly available.

Btw currently I am using the keywords used in the lang compiler.
https://github.com/ballerina-platform/ballerina-lang/blob/master/ballerina-shell/modules/shell-cli/src/main/resources/command.keywords.txt

Reply:
As far as I know there is no such api

Reply:
In that case, is it possible to fetch this file and read the keywords? Or better to request an API from the lang team so that the others can use it.
Maintaining the list of keywords in multiple places will result in issues in the future.

Reply:
Found a lang API which is publicly available. Fixed this in a76184d

`io.ballerina.compiler.syntax.tree.SyntaxInfo.isKeyword(String text)`